services/execution_layer.go

```diff
@@ -34,14 +36,60 @@ func latestBlockUpdater(wg *sync.WaitGroup) {
 	}
 }
 
-// LatestEth1BlockNumber will return the latest epoch
+// LatestEth1BlockNumber will return most recent eth1 block number
 func LatestEth1BlockNumber() uint64 {
-	cacheKey := fmt.Sprintf("%d:frontend:latestEth1BlockNumber", utils.Config.Chain.Config.DepositChainID)
+	cacheKey := fmt.Sprintf("%d:frontend:%s", utils.Config.Chain.Config.DepositChainID, latestBlockNumberCacheKey)
 
 	if wanted, err := cache.TieredCache.GetUint64WithLocalTimeout(cacheKey, time.Second*5); err == nil {
 		return wanted
 	} else {
-		logger.Errorf("error retrieving latestEth1BlockNumber from cache: %v", err)
+		logger.Errorf("error retrieving %s from cache: %v", latestBlockNumberCacheKey, err)
 	}
 	return 0
 }
+
+const latestBlockHashRootCacheKey = "latestEth1BlockRootHash"
+
+// headBlockRootHashUpdater updates the hash of the current chain head block
+func headBlockRootHashUpdater(wg *sync.WaitGroup) {
+	firstRun := true
+
+	for {
+		blockRootHash := []byte{}
+		err := db.ReaderDb.Get(&blockRootHash, `
+		SELECT blockroot
+		FROM blocks
+		WHERE status = '1'
```

Comment:
There is no index for just `status` we should use one, or proof that a "multi-index" will cover this case.

Reply:
Here is your proof that the composite index `idx_blocks_status_slot_epoch` is used:

```
EXPLAIN ANALYZE
	SELECT blockroot
	FROM blocks
	WHERE status = '1'
	ORDER BY slot DESC
	LIMIT 1;
```
=>
```
... // some output
   ->  Index Scan Backward using idx_blocks_status_slot_epoch on blocks 
... // some output
```
This works as `idx_blocks_status_slot_epoch` has `status` as its first part, see here:
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/44141630c110fa8284724f18d5e6970fb92ab819/db/migrations/20230612084855_add_missing_indices.sql#L49

Reply:
good